lib/symengine/basic.rb

```diff
@@ -11,8 +11,13 @@ def free_symbols
     end
     
     def abs
-        return SymEngine::abs(self)
+      SymEngine::abs(self)
+    end
+    def to_proc()
+      if self.free_symbols.count > 1
+        raise ArgumentError, "Too many free symbols! Only 1 allowed. Found #{self.free_symbols.count}."
```

Comment:
`.to_proc([x, y])` is indeed weird... while `.to_proc(:x, :y)` for me is clean enough.

In fact, it depends on target usage of `SymEngine`, we should think about it a bit more (yet, for me as rubyist, `func.blah(someshing)` is much, much cleaner than `SymEngine.blah(func, something)` -- if it is definitely "method to convert `func` into blah")

In addition, I still think it is worth to politely ask main SymEngine engineers to clarify the matters about free symbols order ;)